release/tests/test_github_api.py

```diff
@@ -0,0 +1,156 @@
+"""
+Tests for testeng-ci/release/github_api
+"""
+import ddt
+import json
+import re
+import responses
+from responses import GET, POST  # pylint: disable=no-name-in-module
+from unittest import TestCase
+
+from release.tests.aborted import Aborted
+from release.github_api import RequestFailed, GithubApi
+
+
+class EndpointInfo(object):
+    # pylint: disable=too-few-public-methods
+    """ Encapsulates the info we need to test an end point """
+
+    def __init__(
+            self,
+            expected_url,
+            request_maker,
+            success=200,
+            method=GET,
+            request_body=None
+    ):  # pylint: disable=too-many-arguments
+        self.expected_url = expected_url
+        self.request_maker = request_maker
+        self.success = success
+        self.method = method
+        self.request_body = request_body
+
+ENDPOINTS = [
+    EndpointInfo(
+        "https://api.github.com/repos/test-org/test-repo/commits",
+        GithubApi.commits
```

Comment:
I've just never seen that design pattern before, but I think I get it now. `GihubApi.commits(api)` is the same as `api.commits()` (assuming `api` is an instance of `GithubApi`).

My instinct is that keeping the same pattern as the other `EndpointInfo`s is clearer.